### DISCUSSION TITLE
Switching screens in ktx-app

### DISCUSSION THREAD

**Discussion by hpn33** [libktx/ktx#166] [Q&A] [answered]
how switch between screen
set the screen with ktxGame 
but how changed in screen

**Comment by czyzby** [comment]
Not sure what's the issue here.

`KtxGame` code is available [here](https://github.com/libktx/ktx/blob/master/app/src/main/kotlin/ktx/app/game.kt). After you call [`setScreen`](https://github.com/libktx/ktx/blob/master/app/src/main/kotlin/ktx/app/game.kt#L114-L119), the current screen is hidden, and the new screen is resized and shown. Then, `KtxGame` [continuously calls its rendering method](https://github.com/libktx/ktx/blob/master/app/src/main/kotlin/ktx/app/game.kt#L53).

**Comment by hpn33** [comment]
@czyzby 
I'm inside ktxgame
addscreen
addscreen
addscreen
//and then
setscreen

game on a screen, now
and i want change screen in screen class that is set and showing now by code

i do not understand how can change screen out of ktxgame
companion object or static not work because everything is private
event can`t use if statment.
when say
```kotlin

if (currentScreen == AScreen()) setScreent(BScreen()) 

```

**Comment by czyzby** [comment, marked as answer]
You should keep a reference to the original `KtxGame`. For example:

```kotlin
class MyScreen(val game: KtxGame) : KtxScreen {
    // Implement your logic.
}
```

**Comment by hpn33** [comment]
thanks
